From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Todd Kjos <tkjos@google.com>
Date: Fri, 25 Oct 2019 10:48:55 -0700
Subject: ANDROID: gki_defconfig: add FORTIFY_SOURCE, remove SPMI_MSM_PMIC_ARB

Force SPMI_MSM_PMIC_ARB off and enable FORTIFY_SOURCE for GKI

Bug: 142896904
Bug: 134087016
Test: Build/Boot on cuttlefish
Change-Id: Ib6e501f1efee2349d750010a53572dddcb5457c5
Signed-off-by: Todd Kjos <tkjos@google.com>
---
 arch/arm64/configs/gki_defconfig | 2 ++
 arch/x86/configs/gki_defconfig   | 1 +
 2 files changed, 3 insertions(+)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index 80ebf75319a1..10f8ca213484 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -284,6 +284,7 @@ CONFIG_HW_RANDOM_VIRTIO=y
 # CONFIG_I2C_HELPER_AUTO is not set
 CONFIG_SPI=y
 CONFIG_SPMI=y
+# CONFIG_SPMI_MSM_PMIC_ARB is not set
 CONFIG_PINCTRL_AMD=y
 CONFIG_POWER_AVS=y
 CONFIG_POWER_RESET_HISI=y
@@ -449,6 +450,7 @@ CONFIG_SECURITY_PERF_EVENTS_RESTRICT=y
 CONFIG_SECURITY=y
 CONFIG_SECURITY_NETWORK=y
 CONFIG_HARDENED_USERCOPY=y
+CONFIG_FORTIFY_SOURCE=y
 CONFIG_SECURITY_SELINUX=y
 CONFIG_CRYPTO_ADIANTUM=y
 CONFIG_CRYPTO_MD4=y
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
index 949c449b9d1f..306ad5331a48 100644
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -384,6 +384,7 @@ CONFIG_SECURITY_PERF_EVENTS_RESTRICT=y
 CONFIG_SECURITY=y
 CONFIG_SECURITY_NETWORK=y
 CONFIG_HARDENED_USERCOPY=y
+CONFIG_FORTIFY_SOURCE=y
 CONFIG_SECURITY_SELINUX=y
 CONFIG_CRYPTO_ADIANTUM=y
 CONFIG_CRYPTO_SHA256_SSSE3=y
